### DISCUSSION TITLE
fsinfo64 takes long time to complete - is this normal?

### DISCUSSION THREAD

**Discussion by JoeEarlam** [earlephilhower/arduino-pico#2076] [Q&A] [answered]
Hey, I've been playing with the SD cardinfo example trying to get card size and usage. If I run the following:
```
FSInfo64 fs_info;
SDFS.info64(fs_info);
```
It takes almost eight seconds to complete with a 2gb card, and 22 seconds on an 8gb card. Is this normal? If so, is there a way to grab the fsinfo struct without blocking my code for 20+ seconds?

Thanks!

**Comment by earlephilhower** [comment, marked as answer]
This is down to whatever the SDFat library is doing behind the scenes and not really in our control.  You should try increasing the SPI frequency since there's probably lots of reading FAT structures from dozens of 512b sectors on flash over 1-bit SPI.

**Reply by JoeEarlam** [reply]
I did some digging (and installed the upstream SDFat library), I have now clocked my SD card faster, and also defined "MAINTAIN_FREE_CLUSTER_COUNT 1" before including SdFat.h, after the first call it now takes 68mS to return the free clusters count which is much more acceptable.

Thanks!
